linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/CoreWasmLib.scala

```diff
@@ -616,6 +626,119 @@ object CoreWasmLib {
     newFunctionBuilder(functionID, OriginalName(functionID.toString()))
   }
 
+  private def genBoxBoolean()(implicit ctx: WasmContext): Unit = {
+    val fb = newFunctionBuilder(genFunctionID.box(BooleanRef))
+    val xParam = fb.addParam("x", Int32)
+    fb.setResultType(RefType.any)
+
+    fb += GlobalGet(genGlobalID.bTrue)
+    fb += GlobalGet(genGlobalID.bFalse)
+    fb += LocalGet(xParam)
+    fb += Select(List(RefType.any))
+
+    fb.buildAndAddToModule()
+  }
+
+  private def genBoxInt()(implicit ctx: WasmContext): Unit = {
+    val fb = newFunctionBuilder(genFunctionID.box(IntRef))
+    val xParam = fb.addParam("x", Int32)
+    fb.setResultType(RefType.any)
+
+    // Test if the most significant bit is necessary: (x ^ (x << 1)) & 0x80000000
```

Comment:
[note]
- from `0 ~ 2^30-1`:  the first and second MSBs are 0
- from `-2^30 ~ -1`: the first and second MSBs are 1
- Otherwise, the first and second MSBs are the same -> MSB of the masked bit is 1